jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java

```diff
@@ -164,8 +172,19 @@ private int getAndSetProperty(final String key, final int defaultValue) {
 
   /** Will determine if parsed config is complete, setting any applicable values and defaults. */
   void validate() {
-    if (isBlank(serviceUrl)) {
-      throw new ConfigurationException(SERVICE_URL + " must be specified.");
+    if (isBlank(serviceUrl) && !registrySsl) {
+      throw new ConfigurationException(
+          SERVICE_URL + " must be specified unless " + REGISTRY_SSL + " is set to true.");
+    }
+
+    if (isBlank(hostName) && registrySsl) {
+      throw new ConfigurationException(
+          JMX_HOSTNAME + " must be specified unless " + REGISTRY_SSL + " is set to false.");
```

Comment:
```suggestion
          JMX_HOSTNAME + " must be specified if " + REGISTRY_SSL + " is set to true.");
```

I don't like the idea of splitting these values away from the service URL, but this is clearer wording to me.